lib/active_model/serializer/adapter/json_api/pagination_links.rb

```diff
@@ -0,0 +1,52 @@
+module ActiveModel
+  class Serializer
+    class Adapter
+      class JsonApi < Adapter
+        class PaginationLinks
+          FIRST_PAGE = 1
+
+          attr_reader :collection
+
+          def initialize(collection)
+            raise_unless_any_gem_installed
+            @collection = collection
+          end
+
+          def page_links
+            build_links
+          end
+
+          private
+
+          def build_links
+            pages_from.each_with_object({}) do |(key, value), hash|
+              hash[key] = "?page=#{value}&per_page=#{collection.size}"
+            end
+          end
+
+          def pages_from
+            return {} if collection.total_pages == FIRST_PAGE
+
+            {}.tap do |pages|
+              unless collection.current_page == FIRST_PAGE
+                pages[:first] = FIRST_PAGE
+                pages[:prev]  = collection.current_page - FIRST_PAGE
+              end
+
+              unless collection.current_page == collection.total_pages
+                pages[:next] = collection.current_page + FIRST_PAGE
+                pages[:last] = collection.total_pages
+              end
+            end
+          end
+
+          def raise_unless_any_gem_installed
+            return if defined?(WillPaginate) || defined?(Kaminari)
```

Comment:
See my (too long) comment over in https://github.com/rails-api/active_model_serializers/pull/1041/files#r37346434 but if we have

```ruby
      def initialize(collection)
            raise_unless_any_gem_installed
            @collection = collection
          end
```

rather than checking for WillPaginate or Kaminari, couldn't we check `collection.respond_to?(:total_pages) && collection.respond_to?(:current_page)` since that appears to be the extent of the paginating libs interface we need? (well, we can assume  `collection.size` works correctly if the other two methods are available.)

We'd still have an error message, but it would be that the resource needs these two methods, and we test against both these libraries which support it.